features/device-config/rest/src/main/java/org/opennms/features/deviceconfig/rest/impl/DefaultDeviceConfigRestService.java

```diff
@@ -56,42 +88,35 @@ public Response getDeviceConfigs(
             String orderBy,
             String order,
             Integer ipInterfaceId,
+            String configType,
             Long createdAfter,
             Long createdBefore
     ) {
-        var criteriaBuilder = new CriteriaBuilder(DeviceConfig.class);
+        var criteria = getCriteria(limit, offset, orderBy, order, ipInterfaceId, configType, createdAfter, createdBefore);
```

Comment:
Maybe we need another kind of query that groups by (ipAddressId, configType) and select the maximum of each group by the lastUpdated field.

Reply:
I agree - I'm just not quite sure what data we exactly need, but hope that can be clarified soon (i.e. in next few days or week).